srml/consensus/src/lib.rs

```diff
@@ -14,7 +14,89 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
 
-//! Consensus module for runtime; manages the authority set ready for the native code.
+//! ## Overview
+//! 
+//! The consensus module manages the authority set for the native code. It provides support for reporting offline behavior among validators and logging changes in the validator authority set.
+//!
+//! ## Interface
+//! 
+//! ### Dispatchable
+//! 
+//! The `Call` enum is documented [here](https://crates.parity.io/srml_consensus/enum.Call.html).
+//! 
+//! - `report_misbehavior` - Report some misbehaviour. The origin of this call must be signed.
+//! - `note_offline` - Note the previous block's validator missed their opportunity to propose a block. The origin of this call must be an inherent.
+//! - `remark` - Make some on-chain remark. The origin of this call must be signed.
+//! - `set_heap_pages` - Set the number of pages in the WebAssembly environment's heap.
+//! - `set_code` - Set the new code.
+//! - `set_storage(Vec<KeyValue>)` - Set some items of storage.
+//! 
+//! ### Public
+//! 
+//! See the [module](https://crates.parity.io/srml_consensus/struct.Module.html) for details on publicly available functions.
+//! 
+//! **Note:** When using the publicly exposed functions, you (the runtime developer) are responsible for implementing any necessary checks (e.g. that the sender is the signer) before calling a function that will affect storage.
+//! 
+//! ## Usage
+//! 
+//! ### Prerequisites
+//! 
+//! To use a functionality from the `consensus` module, call the specific Trait or function that you are invoking from the module with the Rust syntax for doing so, ie
```

Comment:
We shouldn't tell them to use proper Rust syntax when they are using Rust :)
```suggestion
//! To use functionality from the `consensus` module, implement the specific Trait or function that you are invoking from the module:
```